### PR TITLE
don't log ES delivery lag for instances that aren't running

### PR DESCRIPTION
**Pull request by rgarcia**
This should fix the ES delivery lag detector

### CODE REVIEW

**Comment by rgarcia** [comment]
had to add aws-sdk-go dep so I converted to glide. Operative commit to review is https://github.com/Clever/log-monitor-es/pull/18/commits/8a31215ee36537c54102a884ff9e7d14e472b6f2

**Comment by rgarcia** [comment]
@nathanleiby also made https://github.com/Clever/infra/pull/627 and assigned you. no rush

File: main.go [hunk 1]
```diff
@@ -126,6 +178,20 @@ func main() {
...
+		// remove data for instances that aren't running
+		for hostname := range timestamps {
+			if strings.HasPrefix(hostname, "ip-") {
+				// convert ES hostnames of the form ip-10-0-0-1 to 10.0.0.1
+				ip := strings.Replace(strings.TrimPrefix("ip-", hostname), "-", ".", -1)
+				running, err := ec2ip.IsRunning(ip)
+				if err != nil {
+					kvlog.ErrorD("ec2-ip-check", kv.M{"error": err.Error()})
+				} else if !running {
+					delete(timestamps, hostname)
```
Comment (nathanleiby) [inline main.go:190]: not sure if deleting while iterating over the slice has predictable behavior. 

to be sure, you could allocate another slice and copy valid elements to it

**Review by nathanleiby — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (rgarcia) [reply main.go:190]: https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-golang-map-within-a-range-loop

Comment (nathanleiby) [reply main.go:190]: 👍 til (I googled a different stackoverflow post that was less decisive :P)

**Pull request merged**